learningresources/forms.py

```diff
@@ -35,11 +37,18 @@ class Meta:
     # The ModelForm.save() accepts "commit" and this doesn't, because
     # we always set commit=False then add the user because created_by is
     # not part of the form, and shouldn't be.
+    @transaction.atomic
     def save(self, user):
         """
         Save a newly-created form.
         """
         repo = super(RepositoryForm, self).save(commit=False)
+        slug = slugify(repo.name)
+        count = 1
+        while Repository.objects.filter(slug=slug).exists():
```

Comment:
Is this typically how slugs are generated? Looks fine to me, just double checking that there's no awesome Django `generate_unique_slug()` function